### ISSUE TITLE
Failing to run docker.

### ISSUE THREAD

**Issue by cyberfox** [webbertakken/unity-actions#37]
Greetings,
I was having problems with my attempt to set up a Unity build flow, with the error:
```
The process '/usr/bin/docker' failed with exit code 1
```
I couldn't figure out what I was doing wrong, so I cloned this repository (unity-actions) and added a UNITY_LICENSE secret to my clone, enabled actions, and committed a minor change.

Same problem.

Is this Unity Build Action currently working for you?

This is the build results for my own build, and the unity-actions repo behaved similarly.

<img width="606" alt="Screen Shot 2020-01-26 at 6 10 38 AM" src="https://user-images.githubusercontent.com/5918/73136449-a7f22d00-4002-11ea-8c12-9a1d8f2cf124.png">

**Comment by cyberfox** [comment]
Okay; I misunderstood, and probably because it was very late at night (and I could not figure out how to see the actual log at the time).  The actual problem I'm having is:
```log
DisplayProgressbar: Unity license
[UnityConnectServicesConfig] config is NOT valid, switching to default
[0126/085839:ERROR:browser_main_loop.cc(161)] Running without the SUID sandbox! See https://code.google.com/p/chromium/wiki/LinuxSUIDSandboxDevelopment for more information on developing with the sandbox on.
[0126/085839:ERROR:sandbox_linux.cc(308)] InitializeSandbox() called with multiple threads in process gpu-process
Cancelling DisplayDialog: Failed to activate/update license. Missing or bad username and password. Please try again using valid credentials or contact support@unity3d.com
This should not be called in batch mode.
 
(Filename: ./Editor/Platform/Linux/EditorUtility.cpp Line: 274)
```
I don't have a place to put a username and password, and there's no directions to do it in the projects, so I'm not sure what I'm missing.

**Comment by webbertakken** [comment]
Hey there @cyberfox,

Thank you for reporting your problems setting up the action. 

> I don't have a place to put a username and password, and there's no directions to do it in the projects, so I'm not sure what I'm missing.

Unity builder has a [readme](https://github.com/webbertakken/unity-builder) that includes an ENV variable called `UNITY_LICENSE`.

Unity Activate also has a [readme](https://github.com/marketplace/actions/unity-activate) that shows both how to activate Unity Personal License and Unity Professional License. You can follow the part that applies to you. 

No that in order to retrieve the value to use for `UNITY_LICENSE` secret you need to request an activation file (only once). Read how to do that [here](https://github.com/webbertakken/unity-request-manual-activation-file)

> Is this Unity Build Action currently working for you?

Yes, every version is tested in the actions own workflow before any merge. You can find this comprehensive workflow here: https://github.com/webbertakken/unity-builder/blob/master/.github/workflows/main.yml 
_(please be so kind to not copy my personal license, this is required there for pull requests)_

In any case could you please also provide the configuration you have used to configure your workflow?

**Comment by cyberfox** [comment]
Greetings,
The examples and unity-actions use `webbertakken/unity-activate@v1` as the activation step to use, but it won't currently work for me without `webbertakken/unity-activate@v1.1`.

So I got the activation step working, but the build is still failing, so I'm working on figuring out what needs to happen for those.

I didn't intend to suggest your code wasn't tested, and I'm very sorry it came across that way.  I was originally trying to figure out if GitHub's servers were having trouble, and maybe they'd started since your last commit.

This is my configuration right now; it gets past Request Activation, but then fails in _Build for StandaloneOSX_ and _Build for StandaloneWindows64_.
```yml
name: Unity - Builder

on:
  pull_request: {}
  push: { branches: [master] }

env:
  UNITY_LICENSE: ${{ secrets.UNITY_LICENSE }}

jobs:
  requestActivation:
    name: Request activation ✔
    runs-on: ubuntu-latest
    steps:
      # Checkout repository
      - name: Checkout repository
        uses: actions/checkout@v1

      # Activate Unity
      - name: Unity - Activate
        uses: webbertakken/unity-activate@v1.1
        env:
          UNITY_LICENSE: ${{ secrets.UNITY_LICENSE }}

  buildForSomePlatforms:
    name: Build for ${{ matrix.targetPlatform }} on version ${{ matrix.unityVersion }}
    runs-on: ubuntu-latest
    strategy:
      fail-fast: false
      matrix:
        projectPath:
          - .
          # path/to/your/project
        unityVersion:
          - 2019.2.14f1
        targetPlatform:
          - StandaloneOSX # Build a macOS standalone (Intel 64-bit).
          - StandaloneWindows64 # Build a Windows 64-bit standalone.
    steps:
      - uses: actions/checkout@v1
      - uses: webbertakken/unity-builder@v0.8
        with:
          projectPath: ${{ matrix.projectPath }}
          unityVersion: ${{ matrix.unityVersion }}
          targetPlatform: ${{ matrix.targetPlatform }}
      - uses: actions/upload-artifact@v1
        with:
          name: Build
          path: build
```

**Comment by cyberfox** [comment]
Fwiw, I have tried `webbertakken/unity-builder@master`, `@v0.8` and `@d5552ea` and none of them seem to recognize the activation like `unity-activate@v1.1` does.

**Comment by webbertakken** [comment]
I'll investigate this issue when I get the time for it.

**Comment by webbertakken** [comment]
> This is the build results for my own build, and the unity-actions repo behaved similarly.###

Unity Actions fork is failing because your license does not exist or is not in valid format. I updated the repo to hold a static license. Of you update your fork with these changes it should work for the fork.

As for your private repo, be sure that you request the activation file using the [Unity - Request Activation File](https://github.com/webbertakken/unity-request-manual-activation-file) and activate that file. This should yield your Unity License File. Note that this license is machine bound and only work on GitHub Actions.

I'm missing the output of your private repos workflow. Please note we'll need pretty specific details because I have not been able to reproduce this issue. Many people have it working.

**Comment by cyberfox** [comment]
Yes, I ran 'Unity - Request Activation File' and generated an .alf, uploaded it to Unity for manual activation, giving me a .ulf file.  I copied that file (using pbcopy), removed the prior `UNITY_LICENSE` and pasted it into the `UNITY_LICENSE` secret.

I then updated the main.yml file with just the `requestActivation`, and `buildForSomePlatforms` jobs.

`requestActivation` works but `buildForSomePlatforms` doesn't.  And it's unclear _why_ it's failing.

My current main.yml looks like this:
```yml
name: Unity - Builder

on:
  pull_request: {}
  push: { branches: [master] }

env:
  UNITY_LICENSE: ${{ secrets.UNITY_LICENSE }}

jobs:
  requestActivation:
    name: Request activation ✔
    runs-on: ubuntu-latest
    steps:
      # Checkout repository
      - name: Checkout repository
        uses: actions/checkout@v1

      # Activate Unity
      - name: Unity - Activate
        uses: webbertakken/unity-activate@v1.1
        env:
          UNITY_LICENSE: ${{ secrets.UNITY_LICENSE }}

  buildForSomePlatforms:
    name: Build for ${{ matrix.targetPlatform }} on version ${{ matrix.unityVersion }}
    runs-on: ubuntu-latest
    strategy:
      fail-fast: false
      matrix:
        projectPath:
          - power-offense
          # path/to/your/project
        unityVersion:
          - 2019.2.14f1
        targetPlatform:
          - StandaloneOSX # Build a macOS standalone (Intel 64-bit).
          - StandaloneWindows64 # Build a Windows 64-bit standalone.
    steps:
      - uses: actions/checkout@v1
      - uses: webbertakken/unity-activate@v1.1
      - uses: webbertakken/unity-builder@d5552ea
        with:
          projectPath: ${{ matrix.projectPath }}
          unityVersion: ${{ matrix.unityVersion }}
          targetPlatform: ${{ matrix.targetPlatform }}
      - uses: actions/upload-artifact@v1
        with:
          name: Build
          path: build
```

I have changed my repository to have a `power-offense` directory in the top with all the project under it, where I used to have `.` as my `projectPath` and I wondered if that might be what was giving me trouble, but the logs are nearly identical.  In fact, the logs for the last half dozen build attempts are almost all identical.

[po_logs_14.zip](https://github.com/webbertakken/unity-actions/files/4116288/po_logs_14.zip)

**Comment by webbertakken** [comment]
So what I can see in `5_Run webbertakkenunity-builder@d5552ea.txt` are the following lines for note:

```
2020-01-27T08:52:52.8997564Z [command]/usr/bin/docker run --workdir /github/workspace --rm --env UNITY_LICENSE --env UNITY_EMAIL --env UNITY_PASSWORD --env UNITY_SERIAL --env UNITY_VERSION=2019.2.14f1 --env PROJECT_PATH=power-offense --env BUILD_TARGET=StandaloneOSX --env BUILD_NAME=StandaloneOSX --env BUILD_PATH=build/StandaloneOSX --env BUILD_FILE=StandaloneOSX --env BUILD_METHOD= --env HOME=/github/home --env GITHUB_REF --env GITHUB_SHA --env GITHUB_REPOSITORY --env GITHUB_ACTOR --env GITHUB_WORKFLOW --env GITHUB_HEAD_REF --env GITHUB_BASE_REF --env GITHUB_EVENT_NAME --env GITHUB_WORKSPACE=/github/workspace --env GITHUB_ACTION --env GITHUB_EVENT_PATH --env RUNNER_OS --env RUNNER_TOOL_CACHE --env RUNNER_TEMP --env RUNNER_WORKSPACE --volume /var/run/docker.sock:/var/run/docker.sock --volume /home/runner/work/_temp/_github_home:/github/home --volume /home/runner/work/_temp/_github_workflow:/github/workflow --volume /home/runner/work/power-offense/power-offense:/github/workspace unity-builder:2019.2.14f1-mac
2020-01-27T08:52:53.2784214Z Requesting activation (personal license)
2020-01-27T08:52:53.4667187Z Activation complete.
2020-01-27T08:52:53.4669501Z Using project path "/github/workspace/power-offense".
```

This is the activation step that's build-in to builder. It says that the license looks like a correct license and tells us that license activation is successful. 

However when using the activated license to build the project, it turns out that the license is not for this machine or not formatted correctly; as the integrity hashes are different.

```
2020-01-27T08:52:53.6267646Z Initiating legacy licensing module
2020-01-27T08:52:53.8891919Z 
2020-01-27T08:52:53.8894617Z LICENSE SYSTEM [2020127 3:52:53] Next license update check is after 2020-01-28T08:44:26
2020-01-27T08:52:53.8895629Z 
2020-01-27T08:52:53.8895987Z 
2020-01-27T08:52:53.8897003Z LICENSE SYSTEM [2020127 3:52:53] 2c6b02a61863466ba72df398410ac5c6 != d39b8e2f4d364b2e98b06afa0c6e08c5
2020-01-27T08:52:53.8897744Z 
2020-01-27T08:52:53.8898082Z 
2020-01-27T08:52:53.8899088Z LICENSE SYSTEM [2020127 3:52:53] 2c6b02a61863466ba72df398410ac5c6 != d39b8e2f4d364b2e98b06afa0c6e08c5
2020-01-27T08:52:53.8899981Z 
2020-01-27T08:52:53.8900828Z DisplayProgressbar: Unity license
2020-01-27T08:52:54.3899719Z [UnityConnectServicesConfig] config is NOT valid, switching to default
```

Looks like one of two things is happening:
- you didn't copy the contents of your `.ulf` file correctly
- since you're using a different version than `2019.2.11f1` (which is used to request the activation file in [request activation file](https://github.com/webbertakken/unity-request-manual-activation-file) action, the config is somehow invalid (in which case this is indeed a bug)

To verify which of the two applies, could you try the same workflow setting `unityVersion` to `2019.2.11f1`? If it then works it's indeed a bug in version mismatch.

**Comment by cyberfox** [comment]
I am developing using 2019.2.14f1 (it's in my build, as `unityVersion` up above) for my project; is that what you mean?

Do I have to use specifically 2019.2.11f1 for my project in order to do this?  I did notice that the license activation process generates one for 2019.2.11f1; I take it if I want to use a different version I need to fork your repos and change them to that version?

**Comment by webbertakken** [comment]
> I am developing using 2019.2.14f1 (it's in my build, as `unityVersion` up above) for my project; is that what you mean?

In your project is fine. But could you try the workflow with 2019.2.11f1, just to see if that is the problem?

> Do I have to use specifically 2019.2.11f1 for my project in order to do this? I did notice that the license activation process generates one for 2019.2.11f1; I take it if I want to use a different version I need to fork your repos and change them to that version?

No. The idea is that we support all versions. It's just a lot of work to develop everything (you're always free to send PRs). Unity Builder has already been converted to use all versions, but the rest of the actions haven't. hence https://github.com/webbertakken/unity-actions/issues/33.

**Comment by cyberfox** [comment]
Yeah; as soon as I read that, I tried a build with my `unityVersion` set to 2019.2.11f1 and it's gotten a lot farther (it's ~21 minutes in so far).  It's doing a ton of asset stuff right now, so we'll see how that comes out.  Thanks very much; I'll update when it's done.

I'm happy to send PRs once I've figured out how everything works.  If I can figure out how to make `Unity - Request Activation File` take a unityVersion, so you can get an activation for any version, that'd be pretty high up there.  It all needs to hang together in lockstep from what I can see, though.

**Comment by webbertakken** [comment]
You may also be interested in this (for now undocumented) caching possiblity https://github.com/webbertakken/unity-builder/pull/27/commits/978a7a274a3ed0c4f86d793d71e5978a5ecc677b

**Comment by cyberfox** [comment]
It built successfully, but it didn't include any of my texture images, models, etc...  Not sure what went wrong there.  At first I thought it was an LFS failure, but other binary objects seem to have been preserved.  I'll look into it more tomorrow.  I've included the logs here, in case you're interested in seeing what it ended up doing.

[po_logs_15.zip](https://github.com/webbertakken/unity-actions/files/4117101/po_logs_15.zip)

**Comment by webbertakken** [comment]
That probably has everything to do with that your project is built on a high version of unity and we're now building with a lower version.

I'll create a ticket for the request license activation file action

**Comment by webbertakken** [comment]
Progress can be tracked in above issue. 

Please feel free to create a new issue if your texture problems are not version mismatch related.

**Comment by millonf** [comment]
Hello,

I'm having the same issue as cyberfox had, with the mismatch of Unity versions:

```
LICENSE SYSTEM [202027 19:24:19] Next license update check is after 2020-02-09T00:04:32


LICENSE SYSTEM [202027 19:24:19] 0c23597de9454b8599ffcec97c54a2e5 != d39b8e2f4d364b2e98b06afa0c6e08c5


LICENSE SYSTEM [202027 19:24:19] 0c23597de9454b8599ffcec97c54a2e5 != d39b8e2f4d364b2e98b06afa0c6e08c5

DisplayProgressbar: Unity license
[UnityConnectServicesConfig] config is NOT valid, switching to default
```

The reason being I'm trying to build on `2019.3.0f6` but the file I receive from the activation job is `Unity_v2019.2.11f1.alf` , containing inside `<ClientProvidedVersion Value="2019.2.11f1"/>`.

I followed all this thread and #30, I verified https://hub.docker.com/r/gableroux/unity3d/tags and it has images for the version, I'm lost.

Here's what I do:

Activation.yml
```yml
name: Acquire activation file
on: [push]
jobs:
  activation:
    name: Request manual activation file 🔑
    runs-on: ubuntu-latest
    steps:
    # Request manual activation file
      - name: Request manual activation file
        id: getManualLicenseFile
        uses: webbertakken/unity-request-manual-activation-file@v1
        with:
          unityVersion: 2019.3.0f6
    # Upload artifact (Unity_v20XX.X.XXXX.alf)
      - name: Expose as artifact
        uses: actions/upload-artifact@v1
        with:
          name: ${{ steps.getManualLicenseFile.outputs.filePath }}
          path: ${{ steps.getManualLicenseFile.outputs.filePath }}
```

part of Build.yml
```yml
jobs:

  requestActivation:
    name: Request activation ✔
    runs-on: ubuntu-latest
    steps:
      # Checkout repository
      - name: Checkout repository
        uses: actions/checkout@v1

      # Activate Unity
      - name: Activate Unity
        uses: webbertakken/unity-activate@v1.2
        env:
          UNITY_LICENSE: ${{ secrets.UNITY_LICENSE }}
        with:
          unityVersion: 2019.3.0f6  


#  requestManualActivationFile:
#    runs-on: ubuntu-latest
#    steps:
#      # Checkout repository
#      - name: Checkout repository
#        uses: actions/checkout@v1

      # Request manual activation file
#      - name: Request manual activation file
#        uses: webbertakken/unity-request-manual-activation-file@master
#        id: getManualLicenseFile

      # Upload artifact (Unity_v20XX.X.XXXX.alf)
#      - name: Expose as artifact
#        uses: actions/upload-artifact@v1
#        with:
#          name: Manual Activation File
#          path: ${{ steps.getManualLicenseFile.outputs.filePath }}

  buildForSomePlatforms:
    name: Build for ${{ matrix.targetPlatform }} on version ${{ matrix.unityVersion }}
    runs-on: ubuntu-latest
    strategy:
      fail-fast: false
      matrix:
        projectPath:
          - Source/Game/
        unityVersion:
          - 2019.3.0f6
        targetPlatform:
          - StandaloneWindows # Build a Windows standalone.
          - StandaloneWindows64 # Build a Windows 64-bit standalone.
```

@webbertakken can you please have a look ?
I am specifying the versions correctly, the identification works but it's just the wrong version I get, even though I specify `unityVersion: 2019.3.0f6 `

**Comment by webbertakken** [comment]
Hi @millonf,

The reason is that you're not using the latest version of request activation file Action. https://github.com/webbertakken/unity-request-manual-activation-file/releases/tag/v1.1

Assuming you followed the readme, its actually my fault because I didn't update it to reflect the latest version. 

Let me know if that works

**Comment by millonf** [comment]
That was my issue, thanks a lot for the fast answer.
I was indeed following the readme, and forgot you created a new release which was 1**.1** .
